thevoid/connection.cpp

```diff
@@ -542,12 +618,18 @@ void connection<T>::print_access_log()
 		m_access_status,
 		m_access_received,
 		m_access_sent,
-		delta);
+		delta,
+		timespec_to_usec(m_receive_time),
+		timespec_to_usec(m_send_time));
 }
 
 template <typename T>
-void connection<T>::handle_read(const boost::system::error_code &err, std::size_t bytes_transferred)
+void connection<T>::handle_read(const boost::system::error_code &err, std::size_t bytes_transferred,
+		struct timespec start_time)
 {
+	auto read_time = gettime_now() - start_time;
+	m_receive_time += read_time;
```

Comment:
It is better to split `m_receive_time` into 2 values:
1. the time it took from calling `async_read_some` until the first `handle_read` is called - **latency time**
2. the time it took from the first until the last `handle_read` are called without time spent in the `handle_read` - **receive time**

Something similar is available from `curl_easy_getinfo()` (http://curl.haxx.se/libcurl/c/curl_easy_getinfo.html#TIMES).